### PR TITLE
Cleaning up using packages page

### PR DESCRIPTION
**Pull request by sfshaza2**
Fixes https://github.com/flutter/website/issues/1906

cc @domesticmouse

### CODE REVIEW

**Review by domesticmouse — APPROVED** [review]

File: src/development/packages-and-plugins/using-packages.md [hunk 1]
```diff
@@ -39,17 +39,29 @@ an app without having to develop everything from scratch.
...
-To add assets, images or fonts,
+To add assets, images, or fonts,
 whether stored in files or packages,
 see [Adding assets and images][].
 
+
+[Adding assets and images]: {{site.url}}/development/ui/assets-and-images
+[`battery`]: {{site.pub-pkg}}/battery
+[developing packages]: {{site.url}}/development/packages-and-plugins/developing-packages
+[FlutterFire]: {{site.repo.plugins}}/blob/master/FlutterFire.md
```
Comment (parlough) [inline src/development/packages-and-plugins/using-packages.md:57]:
```suggestion
[FlutterFire]: {{site.github}}/firebase/flutterfire
```

File: src/development/packages-and-plugins/using-packages.md [hunk 2]
```diff
@@ -72,10 +84,19 @@ be a Flutter Favorite, see the
...
 
 You can also browse the packages on pub.dev by filtering
-on [Android plugins][], [iOS plugins][], [web plugins][],
-[Linux plugins][], [Windows plugins][], [macOS plugins][],
+on [Android][], [iOS][], [web][],
+[Linux][], [Windows][], [macOS][],
 or any combination thereof.
 
+[Android]: {{site.pub-pkg}}?q=sdk%3Aflutter+platform%3Aandroid
+[Flutter Favorites]: {{site.pub}}/flutter/favorites
```
Comment (parlough) [inline src/development/packages-and-plugins/using-packages.md:92]: Seems this was the reference earlier:

```suggestion
[Flutter Favorites program]: {{site.pub}}/flutter/favorites
```

**Review by parlough — APPROVED** [review]
Looks good to me, after adding the missing link and updating this one.

File: src/development/packages-and-plugins/using-packages.md [hunk 3]
```diff
@@ -72,10 +84,20 @@ be a Flutter Favorite, see the
...
+[Flutter Favorites program]: https://docs.flutter.dev/development/packages-and-plugins/favorites
```
Comment (parlough) [inline src/development/packages-and-plugins/using-packages.md:93]:
```suggestion
[Flutter Favorites program]: {{site.url}}/development/packages-and-plugins/favorites
```

**Pull request merged**